pkg/local-gadget-manager/local-gadget-manager.go

```diff
@@ -0,0 +1,309 @@
+		gadgetOperation, ok := tracer.factory.Operations()[opname]
+		if !ok {
+			return fmt.Errorf("Unknown operation %q", opname)
+		}
+		tracerNamespacedName := tracer.traceResource.ObjectMeta.Namespace +
+			"/" + tracer.traceResource.ObjectMeta.Name
+		gadgetOperation.Operation(tracerNamespacedName, tracer.traceResource)
+	}
+
+	fmt.Printf("State: %s\n", tracer.traceResource.Status.State)
+	fmt.Printf("OperationError: %s\n", tracer.traceResource.Status.OperationError)
+	fmt.Printf("Output: %s\n", tracer.traceResource.Status.Output)
+
+	return nil
+}
+
+func (l *LocalGadgetManager) Show(name string) error {
+	return l.Operation(name, "")
+}
+
+func (l *LocalGadgetManager) Delete(name string) error {
+	tracer, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("unknown trace")
+	}
+
+	factory, ok := l.traceFactories[tracer.gadget]
+	if !ok {
+		return fmt.Errorf("unknown gadget")
+	}
+	factory.Delete("gadget/" + name)
+	delete(l.tracers, name)
+	return nil
+}
+
+func (l *LocalGadgetManager) PublishEvent(tracerID string, line string) error {
+	name := strings.TrimPrefix(tracerID, "trace_gadget_")
+	t, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("cannot find tracer: unknown tracer %q", tracerID)
+	}
+
+	t.gadgetStream.Publish(line)
+	return nil
+}
+
+func (l *LocalGadgetManager) Stream(name string, stop chan struct{}) (chan string, error) {
+	t, ok := l.tracers[name]
+	if !ok {
+		return nil, fmt.Errorf("cannot find trace %q", name)
+	}
+
+	out := make(chan string)
+
+	ch := t.gadgetStream.Subscribe()
+
+	go func() {
+		if stop == nil {
+			for len(ch) > 0 {
+				line := <-ch
+				out <- line.Line
+			}
+			t.gadgetStream.Unsubscribe(ch)
+			close(out)
+		} else {
+			for {
+				select {
+				case <-stop:
+					t.gadgetStream.Unsubscribe(ch)
+					close(out)
+					return
+				case line := <-ch:
+					out <- line.Line
+				}
+			}
+		}
+	}()
+	return out, nil
+}
+
+func (l *LocalGadgetManager) AddContainer(notif runcfanotify.ContainerEvent) {
+	pid := int(notif.ContainerPID)
+
+	cgroupPathV1, cgroupPathV2, err := containerutils.GetCgroupPaths(pid)
+	if err != nil {
+		return
+	}
+	cgroupPathV2WithMountpoint, _ := containerutils.CgroupPathV2AddMountpoint(cgroupPathV2)
+	cgroupId, _ := containerutils.GetCgroupID(cgroupPathV2WithMountpoint)
+	mntns, err := containerutils.GetMntNs(pid)
+
+	containerDefinition := pb.ContainerDefinition{
+		Id:         notif.ContainerID,
+		Pid:        notif.ContainerPID,
+		Namespace:  "local",
+		Podname:    notif.ContainerID,
+		CgroupPath: cgroupPathV2WithMountpoint,
+		CgroupId:   cgroupId,
+		Mntns:      mntns,
+		CgroupV1:   cgroupPathV1,
+		CgroupV2:   cgroupPathV2,
+	}
+
+	if l.dockercli != nil {
+		filter := filters.NewArgs()
+		filter.Add("id", notif.ContainerID)
+		containers, err := l.dockercli.ContainerList(context.Background(),
+			dockertypes.ContainerListOptions{
+				All:     true,
+				Filters: filter,
+			})
+		if err != nil {
+			fmt.Printf("Couldn't find docker container: %s", err)
```

Comment:
Nothing to be done apart from logging the error? Also, should it use a logger to print as debug message?